### ISSUE TITLE
Leverage Existing Azure Service Clients Using Dependency Injection

### ISSUE THREAD

**Issue by wsugarman** [Xabaril/AspNetCore.Diagnostics.HealthChecks#1371]
**What would you like to be added**: 
Overloads to the various Azure service health check registration methods that re-use an existing service client from the service container.

**Why is this needed**:
Many users of the Azure client SDKs register their clients as a service in their service containers to easily centralize their creation; perhaps the clients are created manually or are loaded from a configuration. For example, users may include something similar to the following in their ASP.NET Core applications:

```csharp
public IConfiguration Configuration { get; }

// ...

services.AddAzureClients(builder => builder.AddBlobServiceClient(Configuration));
```

However, the health checks (at least the ones I've looked at) do not have a registration overload for fetching the client from an `IServiceProvider`. Therefore, users must re-specify all of the settings used to configure their clients.

**Comment by Nisden** [comment]
The clean up that happen in the pull request #1405 broke the changes made to fix #783 as using `GetPropertiesAsync` for checking blob storage access is not possible when using an Azure AD account that only have "Blob Storage Contributor" as permission instead of "Blob Storage Owner"

**Comment by wsugarman** [comment]
> The clean up that happen in the pull request #1405 broke the changes made to fix #783 as using `GetPropertiesAsync` for checking blob storage access is not possible when using an Azure AD account that only have "Blob Storage Contributor" as permission instead of "Blob Storage Owner"

Oh! My mistake. I'll put out another PR to fix this and add a comment so that hopefully the next person doesn't undo it like me.
